### PR TITLE
[Renaming] Improve performance of RenamePropertyRector

### PR DESCRIPTION
**Pull request by samsonasik**
By move get property fetch type before loop.

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed :tada: @TomasVotruba I think it is ready.

File: rules/Renaming/Rector/PropertyFetch/RenamePropertyRector.php [hunk 1]
```diff
@@ -124,7 +125,6 @@ private function processFromPropertyFetch(PropertyFetch $propertyFetch): ?Proper
                 continue;
             }
 
-            $nodeVarType = $this->nodeTypeResolver->getType($propertyFetch->var);
```
Comment (staabm) [inline rules/Renaming/Rector/PropertyFetch/RenamePropertyRector.php:127]: Before this PR the expensive type-resolving was only needed after the cheap name checks succeeded.

I think instead we should leave the check in the loop but do it only after the classlike instanceof check.

Do you have a concrete real world example you are optimizing for?

**Review by staabm — CHANGES_REQUESTED** [review]

File: rules/Renaming/Rector/PropertyFetch/RenamePropertyRector.php (hunk 1 above)
Comment (samsonasik) [reply rules/Renaming/Rector/PropertyFetch/RenamePropertyRector.php:127]: I don't have, probably isset variable inside loop is better..., wdyt?

Comment (staabm) [reply rules/Renaming/Rector/PropertyFetch/RenamePropertyRector.php:127]: you could null it before the loop and onyl lazy init it when its still null (and only after the class like instanceof check)

Comment (samsonasik) [reply rules/Renaming/Rector/PropertyFetch/RenamePropertyRector.php:127]: Ok, Updated 👍

**Pull request merged**